src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,38 @@
+import { useState } from "react";
+
+/**
+ * Custom hook to manage state synchronized with localStorage.
+ * @param key The key in localStorage.
+ * @param initialValue The initial value to use if none is found in localStorage.
+ * @returns A stateful value and a function to update it.
+ */
+export const useLocalStorage = <T>(key: string, initialValue: T): [T, (value: T) => void] => {
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    if (typeof window === "undefined") {
+      return initialValue;
+    }
+    try {
+      const item = window.localStorage.getItem(key);
+      return item ? (JSON.parse(item) as T) : initialValue;
+    } catch (error) {
+      return initialValue;
+    }
```

Comment:
Thanks for the suggested util! This works perfectly for what I needed. Added it in [f45270b](https://github.com/CBIIT/crdc-datahub-ui/pull/476/commits/f45270bcf31cc59d3df2e7175ed2d825d53271c7). I left the try/catch for the storage retrieval, unsure if that would fail if browser permissions are blocking it. If you happen to know, then LMK and I will remove it.